apache-poi/src/test/java/com/baeldung/fastexcel/FastexcelIntegrationTest.java

```diff
@@ -38,7 +43,18 @@ public void whenParsingExcelFile_thenCorrect() throws IOException {
         assertEquals("John Smith", data.get(3).get(0));
         assertEquals("20", data.get(3).get(1));
     }
-
+    
+    @Test
+    public void whenParsingFastExcelFile_thenDetectEmptyRow() throws IOException {
+    	try (FileInputStream file = new FileInputStream(fileLocation); ReadableWorkbook wb = new ReadableWorkbook(file)) {
+            Sheet sheet = wb.getFirstSheet();
+            try (Stream<Row> rows = sheet.openStream()) {
+            	assertEquals(false, fastexcelHelper.isRowEmpty(rows.reduce((first, second) -> second).get()));
```

Comment:
`assertFalse` ?

Reply:
This `reduce` statement is a bit hard to understand

Reply:
I have changed it as per the review comments.